### ISSUE TITLE
Better Native Image Errors

### ISSUE THREAD

**Issue by fniephaus** [oracle/graal#5271]
### TL;DR

We are re-engineering how errors are thrown by GraalVM Native Image to further improve the developer experience.

### Goals

Errors thrown by Native Image can sometimes be difficult to understand or even confusing to users. As an AOT compiler, Native Image is a non-trivial system and currently throws many different types of errors.
To further improve the developer experience, we plan to re-engineer how errors are thrown in Native Image and reported to users.
As part of this, we want to make sure that errors are always actionable, help users resolve build-time and run-time problems quicker, make it easier for them to find related issues and report new issues, and re-categorize and re-classify errors, among other efforts.

As an example, we plan to clearly distinguish between user errors (caused by users; need a clear action) and internal errors (caused by Native Image; need to be reported). As another example, the Native Image build process could generate bug reports automatically that users can copy and paste when creating a new issue.

### Non-Goals

- Focus exclusively on run-time vs. build-time initialization as we are also advancing the programming model
(see https://github.com/oracle/graal/issues/4922)
- Stop using GitHub for error reporting and issue tracking
